src/stats.c

```diff
@@ -451,6 +451,84 @@ mi_msecs_t _mi_clock_end(mi_msecs_t start) {
 }
 
 
+/* -----------------------------------------------------------
+  Histogram of allocations sizes (power of 2)
+----------------------------------------------------------- */
+
+static _Atomic(size_t) mi_hist[MI_SIZE_BITS] = { 0 };
+
+void _mi_histogram_log(size_t size)
+{
+  size_t bucket = MI_SIZE_BITS - 1 - mi_clz(size);
```

Comment:
Why not using `mi_bsr`? That is,
```c
  if (mi_unlikely(size == 0)) return;
  size_t bucket = mi_bsr(size);
```

Reply:
Good idea. I have made these two changes. I think your suggestion for the zero check is necessary unless we were to add another element to the histogram array which also would eliminate a branch. I do a null check in the caller but didn't consider not null pointers with zero size but I believe it does happen when debug is disabled after looking at this conditional statement in `mi_heap_malloc_small`:

https://github.com/microsoft/mimalloc/blob/38a03229c89afbf2722e48c63da696bf11589dee/src/alloc.c#L77-L81

Reply:
I appreciate the feedback from @michaeljclark. Last year, I was exploring `mimalloc` performance issues using [the preliminary work](https://twitter.com/mork9869/status/1387002697665040384). Thus, my proposed change was the part of the workbench.